src/nativeapm/elements/inputs/input.ts

```diff
@@ -0,0 +1,102 @@
+/// <reference path="../../references.ts" />
+
+interface INativeApmInput {
+  getInputElement(): HTMLElement;
+  isValid(): boolean;
+  getValue(): any;
+  setErrorMessage(): void;
+  inputData: NativeApmInputData;
+}
+
+type NativeApmInputData = {
+  key: string;
+  type: string;
+  required: boolean;
+  length: number | null;
+  max_length: number | null;
+  display_name: string;
+  available_values?: Array<{
+    value: string;
+    display_name: string;
+    default: boolean;
+  }>;
+};
+
+/**
+ * ProcessOut module/namespace
+ */
+module ProcessOut {
+  /**
+   * ProcessOut Native APM class for handling input elements based on input type
+   */
+  export class NativeApmInput {
+    /**
+     * Native APM input instance
+     * @type {INativeApmInput}
+     */
+    inputInstance: INativeApmInput;
+
+    /**
+     * Native APM Input constructor
+     */
+    constructor(
+      inputData: NativeApmInputData,
+      theme: NativeApmThemeConfigType,
+      prefilledValue?: string
+    ) {
+      this.inputInstance = this.createInputElement(
+        inputData,
+        theme,
+        prefilledValue
+      );
+    }
+
+    /**
+     * This function returns the input element
+     */
+    public getInputElement() {
+      return this.inputInstance.getInputElement();
+    }
+
+    /**
+     * This function returns the input instance
+     */
+    public getInputInstance() {
+      return this.inputInstance;
+    }
+
+    /**
+     * This function returns the input value
+     */
+    public getInputValue() {
+      return this.inputInstance.getValue();
+    }
+
+    /**
+     * This function creates the input element
+     */
+    private createInputElement(
+      inputData: NativeApmInputData,
+      theme: NativeApmThemeConfigType,
+      prefilledValue: string
+    ) {
+      switch (inputData.type) {
+        case 'email':
+          return new NativeApmEmailInput(inputData, theme, prefilledValue);
+        case 'numeric':
+          if (inputData.length <= 6) {
+            return new NativeApmNumericInput(inputData, theme);
+          } else {
+            return new NativeApmTextInput(inputData, theme, prefilledValue);
```

Comment:
Yeah it is not usual but it seems like this is how Mobile SDK handles it:
https://checkout.atlassian.net/wiki/spaces/~6348713948be855a65ed60a2/pages/5821530336/Native+Alternative+Payment+Method+Mobile#Initiate-the-payment-request

> SDK displays special input field for numeric input field with where length is lower than 6, e.g for blik

Reply:
On mobile we decided to have this limit to make sure UI looks good on smaller devices. If it's possible to handle it differently on web, e.g. by dynamically responding to container size maybe that could be a way to go.

Reply:
Tbh I would leave it as it is right now since on the Web we still need to support Mobile Web and it might look nasty rendering like 15 separate character inputs in two rows